docs/reference/commandline/build.md

```diff
@@ -489,6 +490,49 @@ FROM alpine AS production-env
 $ docker build -t mybuildimage --target build-env .
 ```
 
+### Custom build outputs
+
+By default, a local container image is created from the build result. With the `--output` flag this behavior can be overriden and a custom exporter can be used instead. For example, this is useful for generating local binaries, code generation etc.
```

Comment:
```suggestion
By default, a local container image is created from the build result. With the `--output` flag this behavior can be overriden and a custom exporter can be used instead. For example, custom exporters allow you to export the build artefacts as files on the local filesystem instead of a Docker image, which can be useful for generating local binaries, code generation etc.
```